pynamodb/attributes.py

```diff
@@ -311,6 +313,18 @@ def _set_attributes(self, **attributes: Attribute) -> None:
                 raise ValueError("Attribute {} specified does not exist".format(attr_name))
             setattr(self, attr_name, attr_value)
 
+    def _deserialize(self, attribute_values: Dict[str, Dict[str, Any]]) -> None:
+        """
+        Sets attributes sent back from DynamoDB on this object
+        """
+        self.attribute_values = {}
+        self._set_defaults(_user_instantiated=False)
```

Comment:
This fixes a bug where default values weren't getting set when refreshing or updating models.